inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java

```diff
@@ -187,7 +190,21 @@ public final class Constants {
                     .noDefaultValue()
                     .withDescription(
                             "The format of multiple sink, it represents the real format of the raw binary data");
-
+    public static final ConfigOption<String> PATTERN_PARTITION_MAP =
+            ConfigOptions.key("pattern.partition.map")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Pattern rules and partition map string, " +
```

Comment:
There are already many places using similar formats such as 'key1=value1&key2=value2...', such as dirty data archives and indicators. It is recommended to unify this format.

Reply:
The current format can be automatically converted into a map, eg:

ConfigOptions.key("datasource.partition.map").mapType()